runtime/src/main/java/org/corfudb/runtime/view/stream/FixedBatchSpliterator.java

```diff
@@ -0,0 +1,70 @@
+package org.corfudb.runtime.view.stream;
+
+import java.util.Spliterator;
+import java.util.function.Consumer;
+
+import static java.util.Spliterators.spliterator;
+
+
+/**
+ * This spliterator divides its elements into batches by a
+ * fixed batch size, allowing parallel computation.
+ *
+ * Reference:
+ * https://www.airpair.com/java/posts/parallel-processing-of-io-based-data-with-java-streams
+ */
+public abstract class FixedBatchSpliterator<T> implements Spliterator<T> {
+    private static final int DEFAULT_BATCH = 1 << 6;
+
+    private final int batchSize;
+    private final int characteristics;
+    private long est;
+
+    protected FixedBatchSpliterator(int characteristics, int batchSize, long est) {
+        this.characteristics = characteristics | SUBSIZED;
+        this.batchSize = batchSize;
+        this.est = est;
+    }
+    protected FixedBatchSpliterator(int characteristics, int batchSize) {
+        this(characteristics, batchSize, Long.MAX_VALUE);
+    }
+    protected FixedBatchSpliterator(long est, int characteristics) {
+        this(characteristics, DEFAULT_BATCH, est);
+    }
+    protected FixedBatchSpliterator(int characteristics) {
+        this(characteristics, DEFAULT_BATCH, Long.MAX_VALUE);
+    }
+    protected FixedBatchSpliterator() {
+        this(IMMUTABLE | ORDERED | NONNULL);
+    }
+
+    @Override
+    public Spliterator<T> trySplit() {
+        final HoldingConsumer<T> holder = new HoldingConsumer<>();
+        if (!tryAdvance(holder)) return null;
+        final Object[] a = new Object[batchSize];
+        int j = 0;
+        do a[j] = holder.value; while (++j < batchSize && tryAdvance(holder));
+        if (est != Long.MAX_VALUE) est -= j;
+        return spliterator(a, 0, j, characteristics() | SIZED);
+    }
+
+    @Override
+    public long estimateSize() {
+        return est;
+    }
+
+    @Override
+    public int characteristics() {
+        return characteristics;
+    }
+
+    private static final class HoldingConsumer<T> implements Consumer<T> {
```

Comment:
What is the point of having HoldingConsumer? 
1. why not just use the object?
2. HoldingConsumer brings complexity, developers in the future will have a hard time figuring out what type of an object it holds. Please don't give up on typing, otherwise, we all will struggle with the code

Reply:
> What is the point of having HoldingConsumer?
> 
> 1. why not just use the object?
> 2. HoldingConsumer brings complexity, developers in the future will have a hard time figuring out what type of an object it holds. Please don't give up on typing, otherwise, we all will struggle with the code

It is used to get the object from `tryAdvance`, which does not return the object. But we need those objects to split batch.
And it is borrowed from 'java.util.Spliterators'

Reply:
Why not use Consumer<T> then?
Or at least please use `T value` instead of `Object value`